### PR TITLE
Added tests for untagged and any from mef_eline changes

### PR DESCRIPTION
**Pull request by Alopalao**
Related to [mef_eline](https://github.com/kytos-ng/mef_eline/pull/258)  PR

 ### Summary
Tests including the recent addition to `vlan` available values `untagged` and `any`.

### Local Tests
```
+ python3 -m pytest tests/test_e2e_10_mef_eline.py -k '160 or 165 or 170 or 175 or 180 or 185 or 190 or 195 or 200 or 205' --reruns 2 -r fEr
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-7.2.0, pluggy-1.0.0
rootdir: /tests
plugins: timeout-2.1.0, rerunfailures-10.2
collected 39 items / 29 deselected / 10 selected

tests/test_e2e_10_mef_eline.py ..........                                [100%]
```

### CODE REVIEW

File: tests/test_e2e_10_mef_eline.py [hunk 1]
```diff
@@ -1369,3 +1370,579 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        api_url = KYTOS_API + '/mef_eline/v2/evc/'
+        response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
+        assert response.status_code == 201, response.text
+        data = response.json()
+        assert 'circuit_id' in data
+        time.sleep(10)
+
+        api_url = KYTOS_API + '/flow_manager/v2/flows/00:00:00:00:00:00:00:01'
+        response = requests.get(api_url)
+        data = response.json()["00:00:00:00:00:00:00:01"]["flows"][2]
```
[the remark this answers is not in the document]
Comment (viniarck) [reply tests/test_e2e_10_mef_eline.py:1398]: Maybe a helper function to search for a expected dict flow in a list would be handy to reuse on other tests too.

**Review by viniarck — COMMENTED** [review]
@Alopalao looks great to me. Nicely done covering the cases, including with inter and intra EVCs as well. The spreadsheet that you created facilitated a lot to review (and it'll be linked int the release notes), so much appreciated you creating that as well. 

I'll leave this PR for @italovalcy to finish the review since you both have already been collaborating on a related review.

File: tests/test_e2e_10_mef_eline.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Alopalao) [reply tests/test_e2e_10_mef_eline.py]: Added a new function `get_flow_by_vlan_match`

File: tests/test_e2e_10_mef_eline.py [hunk 2]
```diff
@@ -1369,3 +1370,588 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        h2.cmd('ip link add link %s name vlan_ra type vlan id %s' % (h2.intfNames()[0], ra_vlan))
+        h2.cmd('ip link set up vlan_ra')
+        h2.cmd('ip addr add 100.0.0.2/24 dev vlan_ra')
+        result = h11.cmd('ping -c1 100.0.0.2')
+        assert ', 0% packet loss,' in result
+
+        # Clean up
+        h11.cmd('ip link del vlan_ra')
+        h2.cmd('ip link del vlan_ra')
+        self.net.restart_kytos_clean()
```
Comment (italovalcy) [inline tests/test_e2e_10_mef_eline.py:1487]:
```suggestion
```

File: tests/test_e2e_10_mef_eline.py [hunk 3]
```diff
@@ -1369,3 +1370,588 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        h2.cmd('ip link add link %s name vlan100 type vlan id 100' % (h2.intfNames()[0]))
+        h2.cmd('ip link set up vlan100')
+        h2.cmd('ip addr add 100.0.0.2/24 dev vlan100')
+        result = h11.cmd('ping -c1 100.0.0.2')
+        assert ', 0% packet loss,' in result
+
+        # Clean up
+        h11.cmd('ip link del vlan100')
+        h2.cmd('ip link del vlan100')
+        self.net.restart_kytos_clean()
```
Comment (italovalcy) [inline tests/test_e2e_10_mef_eline.py:1558]:
```suggestion
```

File: tests/test_e2e_10_mef_eline.py [hunk 4]
```diff
@@ -1369,3 +1370,588 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        h11.cmd('ip link set up vlan100')
+        h11.cmd('ip addr add 100.0.0.11/24 dev vlan100')
+        h2.cmd('ip addr add 100.0.0.2/24 dev %s' % (h2.intfNames()[0]))
+        result = h11.cmd('ping -c1 100.0.0.2')
+        assert ', 0% packet loss,' in result
+
+        # Clean up
+        h11.cmd('ip link del vlan100')
+        h2.cmd('ip addr del 100.0.0.2/24 dev %s' % (h2.intfNames()[0]))
+        self.net.restart_kytos_clean()
```
Comment (italovalcy) [inline tests/test_e2e_10_mef_eline.py:1629]:
```suggestion
```

File: tests/test_e2e_10_mef_eline.py [hunk 5]
```diff
@@ -1369,3 +1370,588 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        api_url = KYTOS_API + '/flow_manager/v2/flows/00:00:00:00:00:00:00:02'
+        response = requests.get(api_url)
+        data = response.json()["00:00:00:00:00:00:00:02"]["flows"]
+        untagged_flow = self.get_flow_by_vlan_match(data, 0)
+        assert untagged_flow["match"] == expected[1]["match"]
+        assert untagged_flow["priority"] == expected[1]["priority"]
+        assert untagged_flow["instructions"][0]["actions"] == expected[1]["actions"]
+
+        # Clean up
+        self.net.restart_kytos_clean()
```
Comment (italovalcy) [inline tests/test_e2e_10_mef_eline.py:1689]:
```suggestion
```

File: tests/test_e2e_10_mef_eline.py [hunk 6]
```diff
@@ -1369,3 +1370,588 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        h12.cmd('ip link add link %s name ra_vlan type vlan id %s' % (h12.intfNames()[0], ra_vlan))
+        h12.cmd('ip link set up ra_vlan')
+        h12.cmd('ip addr add 10.1.1.12/24 dev ra_vlan')
+        result = h11.cmd('ping -c1 10.1.1.12')
+        assert ', 0% packet loss,' in result
+
+        # clean up
+        h11.cmd('ip link del ra_vlan')
+        h12.cmd('ip link del ra_vlan')
+        self.net.restart_kytos_clean()
```
Comment (italovalcy) [inline tests/test_e2e_10_mef_eline.py:1740]:
```suggestion
```

File: tests/test_e2e_10_mef_eline.py [hunk 7]
```diff
@@ -1369,3 +1370,588 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        h11, h12 = self.net.net.get('h11', 'h12')
+        h11.cmd('ip addr add 100.1.1.11/24 dev %s' % (h11.intfNames()[0]))
+        h12.cmd('ip addr add 100.1.1.12/24 dev %s' % (h12.intfNames()[0]))
+        result = h11.cmd('ping -c1 100.1.1.12')
+        assert ', 0% packet loss,' in result
+
+        # clean up
+        h11.cmd('ip addr del 100.1.1.11/24 dev %s' % (h11.intfNames()[0]))
+        h12.cmd('ip addr del 100.1.1.12/24 dev %s' % (h12.intfNames()[0]))
+        self.net.restart_kytos_clean()
```
Comment (italovalcy) [inline tests/test_e2e_10_mef_eline.py:1786]:
```suggestion
```

File: tests/test_e2e_10_mef_eline.py [hunk 8]
```diff
@@ -1369,3 +1370,588 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        h12.cmd('ip link add link %s name vlan100 type vlan id 100' % (h12.intfNames()[0]))
+        h12.cmd('ip link set up vlan100')
+        h12.cmd('ip addr add 10.1.1.12/24 dev vlan100')
+        result = h11.cmd('ping -c1 10.1.1.12')
+        assert ', 0% packet loss,' in result
+
+        # clean up
+        h11.cmd('ip link del vlan100')
+        h12.cmd('ip link del vlan100')
+        self.net.restart_kytos_clean()
```
Comment (italovalcy) [inline tests/test_e2e_10_mef_eline.py:1845]:
```suggestion
```

File: tests/test_e2e_10_mef_eline.py [hunk 9]
```diff
@@ -1369,3 +1370,588 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        h11.cmd('ip link set up vlan100')
+        h11.cmd('ip addr add 100.1.1.11/24 dev vlan100')
+        h12.cmd('ip addr add 100.1.1.12/24 dev %s' % (h12.intfNames()[0]))
+        result = h11.cmd('ping -c1 100.1.1.12')
+        assert ', 0% packet loss,' in result
+
+        # clean up
+        h11.cmd('ip link del vlan100')
+        h12.cmd('ip addr del 100.1.1.12/24 dev %s' % (h12.intfNames()[0]))
+        self.net.restart_kytos_clean()
```
Comment (italovalcy) [inline tests/test_e2e_10_mef_eline.py:1909]:
```suggestion
```

File: tests/test_e2e_10_mef_eline.py [hunk 10]
```diff
@@ -1369,3 +1370,588 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        assert any_flow["priority"] == expected[0]["priority"]
+        assert any_flow["instructions"][0]["actions"] == expected[0]["actions"]
+        
+        untagged_flow = self.get_flow_by_vlan_match(data, 0)
+        assert untagged_flow["match"] == expected[1]["match"]
+        assert untagged_flow["priority"] == expected[1]["priority"]
+        assert untagged_flow["instructions"][0]["actions"] == expected[1]["actions"]
+
+        # Clean up
+        self.net.restart_kytos_clean()
```
Comment (italovalcy) [inline tests/test_e2e_10_mef_eline.py:1957]:
```suggestion
```

**Review by italovalcy — APPROVED** [review]
Very well done, @Alopalao. The overall PR looks good to me. I just left a few suggestion to remove kytos restart calls because this will already happen as part of the `setup_method`.

File: tests/test_e2e_10_mef_eline.py [hunk 11]
```diff
@@ -1369,3 +1370,588 @@ def test_155_removing_evc_metadata_persistent(self):
...
+        h11, h2 = self.net.net.get('h11', 'h2')
+        h11.cmd('ip addr add 100.0.0.11/24 dev %s' % (h11.intfNames()[0]))
+        h2.cmd('ip addr add 100.0.0.2/24 dev %s' % (h2.intfNames()[0]))
+        result = h11.cmd('ping -c1 100.0.0.2')
+        assert ', 0% packet loss,' in result
+
+        # Clean up
+        h11.cmd('ip addr del 100.0.0.11/24 dev %s' % (h11.intfNames()[0]))
+        h2.cmd('ip addr del 100.0.0.2/24 dev %s' % (h2.intfNames()[0]))
+        self.net.restart_kytos_clean()
```
[the remark this answers is not in the document]
Comment (Alopalao) [reply tests/test_e2e_10_mef_eline.py]: You are right, deleting all its occurrences in commit [89d4f76e090d3feb6275dfb07494de4b37869739](https://github.com/kytos-ng/kytos-end-to-end-tests/pull/213/commits/89d4f76e090d3feb6275dfb07494de4b37869739)

File: tests/test_e2e_10_mef_eline.py (hunk 2 above)
Comment (Alopalao) [reply tests/test_e2e_10_mef_eline.py]: Done in commit [89d4f76e090d3feb6275dfb07494de4b37869739](https://github.com/kytos-ng/kytos-end-to-end-tests/pull/213/commits/89d4f76e090d3feb6275dfb07494de4b37869739)

File: tests/test_e2e_10_mef_eline.py (hunk 3 above)
Comment (Alopalao) [reply tests/test_e2e_10_mef_eline.py]: Done in commit [89d4f76e090d3feb6275dfb07494de4b37869739](https://github.com/kytos-ng/kytos-end-to-end-tests/pull/213/commits/89d4f76e090d3feb6275dfb07494de4b37869739)

File: tests/test_e2e_10_mef_eline.py (hunk 4 above)
Comment (Alopalao) [reply tests/test_e2e_10_mef_eline.py]: Done in commit [89d4f76e090d3feb6275dfb07494de4b37869739](https://github.com/kytos-ng/kytos-end-to-end-tests/pull/213/commits/89d4f76e090d3feb6275dfb07494de4b37869739)

File: tests/test_e2e_10_mef_eline.py (hunk 5 above)
Comment (Alopalao) [reply tests/test_e2e_10_mef_eline.py]: Done in commit [89d4f76e090d3feb6275dfb07494de4b37869739](https://github.com/kytos-ng/kytos-end-to-end-tests/pull/213/commits/89d4f76e090d3feb6275dfb07494de4b37869739)

File: tests/test_e2e_10_mef_eline.py (hunk 6 above)
Comment (Alopalao) [reply tests/test_e2e_10_mef_eline.py]: Done in commit [89d4f76e090d3feb6275dfb07494de4b37869739](https://github.com/kytos-ng/kytos-end-to-end-tests/pull/213/commits/89d4f76e090d3feb6275dfb07494de4b37869739)

File: tests/test_e2e_10_mef_eline.py (hunk 7 above)
Comment (Alopalao) [reply tests/test_e2e_10_mef_eline.py]: Done in commit [89d4f76e090d3feb6275dfb07494de4b37869739](https://github.com/kytos-ng/kytos-end-to-end-tests/pull/213/commits/89d4f76e090d3feb6275dfb07494de4b37869739)

File: tests/test_e2e_10_mef_eline.py (hunk 8 above)
Comment (Alopalao) [reply tests/test_e2e_10_mef_eline.py]: Done in commit [89d4f76e090d3feb6275dfb07494de4b37869739](https://github.com/kytos-ng/kytos-end-to-end-tests/pull/213/commits/89d4f76e090d3feb6275dfb07494de4b37869739)

File: tests/test_e2e_10_mef_eline.py (hunk 10 above)
Comment (Alopalao) [reply tests/test_e2e_10_mef_eline.py]: Done in commit [89d4f76e090d3feb6275dfb07494de4b37869739](https://github.com/kytos-ng/kytos-end-to-end-tests/pull/213/commits/89d4f76e090d3feb6275dfb07494de4b37869739)

**Pull request merged**